src/parsers/simple.rs

```diff
@@ -1,294 +1,442 @@
-
-use nom::is_digit;
-
-use super::super::{GedcomxDate, Simple, Date, Time, DateTime};
+use nom::branch::alt;
+use nom::bytes::complete::{tag, take_while_m_n};
+use nom::character::complete::one_of;
+use nom::character::is_digit;
+use nom::combinator::{map, opt};
+use nom::sequence::Tuple;
+use nom::IResult;
+
+use super::super::{Date, DateTime, GedcomxDate, Simple, Time};
 use super::approximate;
-use helper::*;
 
-named!(take_4_digits, flat_map!(take!(4), check!(is_digit)));
+fn take_4_digits(i: &str) -> IResult<&str, &str> {
+    take_while_m_n(4, 4, |c: char| c.is_ascii() && is_digit(c as u8))(i)
```

Comment:
The `is_ascii` check let us safely cast a char to a u8, since `is_digit` is unfortunately defined to take a byte rather than a char.